### PR TITLE
Check `hasAISiteGen`, deprecate `canAccessAI`

### PR DESCRIPTION
**Pull request by arunshenoy99**
## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [ ] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments
Please check if this makes sense.

### CODE REVIEW

**Review by diwanshuster — APPROVED** [review]
LGTM

**Review by amartya-dev — APPROVED** [review]

File: includes/SiteGen/SiteGen.php [hunk 1]
```diff
@@ -56,9 +56,10 @@ class SiteGen {
 	private static function check_capabilities() {
 		$capability = new SiteCapabilities();
 
-		$ai_enabled = $capability->get( 'canAccessAI' );
+		$sitegen_enabled = $capability->get( 'hasAISiteGen' );
+		$ai_enabled      = $capability->get( 'canAccessAI' );
```
Comment (MikeHansenMe) [inline includes/SiteGen/SiteGen.php:60]: @arunshenoy99 I think @wpscholar had mentioned wanting to retire canAccessAI in favor of feature specific capabilities.

Comment (amartya-dev) [reply includes/SiteGen/SiteGen.php:60]: yeah, I think the hasAISitegen would be the only capability we will need here.

Comment (arunshenoy99) [reply includes/SiteGen/SiteGen.php:60]: @wpscholar can you confirm whether it's safe to check only the `hasAISiteGen` capability?

**Review by wpscholar — APPROVED** [review]

**Comment by sangeetha-nayak** [comment]
Verified.

**Pull request merged**